docs/apache-airflow/authoring-and-scheduling/deferring.rst

```diff
@@ -213,6 +210,49 @@ This is particularly useful when deferring is the only thing the ``execute`` met
             # We have no more work to do here. Mark as complete.
             return
 
+To enable Dynamic Task Mapping support, you can define ``start_from_trigger`` and ``trigger_kwargs`` in the parameter of "__init__". Note that you don't need to define both of them to use this feature, but you do need to use the exact same parameter name. For example, if you define an argument as ``t_kwargs`` and assign this value to ``self.start_trigger_args.trigger_kwargs``, it will not work.
+
+.. code-block:: python
+
+    from datetime import timedelta
+    from typing import Any
+
+    from airflow.sensors.base import BaseSensorOperator
+    from airflow.triggers.temporal import TimeDeltaTrigger
+    from airflow.utils.context import Context
+
+
+    class WaitTwoHourSensor(BaseSensorOperator):
+        start_trigger_args = StartTriggerArgs(
+            trigger_cls="airflow.triggers.temporal.TimeDeltaTrigger",
+            trigger_kwargs={},
+            next_method="execute_complete",
+            timeout=None,
+        )
+
+        def __init__(
+            self,
+            *args: list[Any],
+            trigger_kwargs: dict[str, Any] | None,
+            start_from_trigger: bool,
+            **kwargs: dict[str, Any],
+        ) -> None:
+            super().__init__(*args, **kwargs)
+            self.start_trigger_args.trigger_kwargs = trigger_kwargs
+            self.start_from_trigger = start_from_trigger
+
+        def execute_complete(self, context: Context, event: dict[str, Any] | None = None) -> None:
+            # We have no more work to do here. Mark as complete.
+            return
+
+These parameters can be mapped using the ``expand`` and ``partial`` methods.
+
+.. code-block:: python
+
+    WaitTwoHourSensor.partial(task_id="transform").partial(start_from_trigger=True).expand(
```

Comment:
High level question here: Shouldn't this be in the operators control, not the DAG authors (at least explicitly like this)?

Reply:
I must be missing some nuance of mapped operators here... Can you give me the tldr?

Reply:
There's one mistake in this example 🤦‍♂️ But I think what we're trying to do is something like https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/dynamic-task-mapping.html#mapping-with-non-taskflow-operators

Reply:
I'm just not sure I see a reason to bring the DAG author into the decision making process. Why should a DAG author have to remember/know to do `.partial(start_from_trigger=True, ...)` all time time? Can't the operator author just set that default?

Reply:
> Can't the operator author just set that default?

Yes, we can. But the operator author might want to make an operator have both modes supported and set it to `False.` If set to `True`, the DAG authors can still use it to turn off this feature

Reply:
I guess I don't see mapped tasks as being fundamentally different than normal tasks in this context - the same ability for an operator author to support both paths applies to them too right? But we document normal tasks with an example that only starts from the triggerer. I'd be inclined to do the same with mapped operators.

Reply:
> I guess I don't see mapped tasks as being fundamentally different than normal tasks in this context - the same ability for an operator author to support both paths applies to them too right?

The reason why it wasn't is that the `__init__` method happens in the worker for the mapped operator. and we expected some DAG authors and operator author would want to use it to decide whether and how to start the execution from trigger

Reply:
Wei and I spent a good amount of time just now talking through this. Ultimately, this feature is only for mapped + deferred (already a smallish subset of tasks), but also will only be usable by operators that _can_ start from the triggerer. The number of tasks that hit all 3 of those is pretty darn small - some very rough math, like somewhere in the neighborhood of .2% of all deferred tasks (obviously varies by specific use cases)!

But if you need it, you need it. I'm not going to hold up this one because of the interface since realistically it'll only be an option for a tiny amount of users anyways.

For posterity: A DAG author being able to control whether it starts in the triggerer or worker is a non-goal imo. Ideally the interface is the same regardless of whether a task starts in the triggerer or worker, but that has technical hurdles.